homeassistant/components/matter/select.py

```diff
@@ -124,6 +140,42 @@ def _update_from_device(self) -> None:
             self._attr_name = desc
 
 
+class MatterDynamicListSelectEntity(MatterEntity, SelectEntity):
+    """Representation of a Matter select entity where the options are dynamically defined in a Matter attribute."""
+
+    entity_description: MatterDynamicListSelectEntityDescription
+
+    async def async_select_option(self, option: str) -> None:
+        """Change the selected option."""
+        option_id = self._attr_options.index(option)
+
+        if TYPE_CHECKING:
+            assert option_id is not None
+        await self.matter_client.write_attribute(
```

Comment:
Once https://github.com/home-assistant/core/pull/136635 is merged, this can be simplified to self.write_attribute